packages/wp-now/src/tests/wp-now.spec.ts

```diff
@@ -144,3 +150,143 @@ test('isWordPressDevelopDirectory returns false for incomplete WordPress-develop
 	expect(isWordPressDevelopDirectory(projectPath)).toBe(false);
 	expect(inferMode(projectPath)).toBe(WPNowMode.INDEX);
 });
+
+describe('Test starting different modes', () => {
+	let tmpExampleDirectory;
+
+	/**
+	 * Copy example directory to a temporary directory
+	 */
+	beforeEach(() => {
+		const tmpDirectory = os.tmpdir();
+		const directoryHash = crypto.randomBytes(20).toString('hex');
+
+		tmpExampleDirectory = path.join(
+			tmpDirectory,
+			`wp-now-tests-${directoryHash}`
+		);
+		fs.ensureDirSync(tmpExampleDirectory);
+		fs.copySync(exampleDir, tmpExampleDirectory);
+	});
+
+	/**
+	 * Remove temporary directory
+	 */
+	afterEach(() => {
+		fs.rmSync(tmpExampleDirectory, { recursive: true, force: true });
+	});
+
+	/**
+	 * Expect that all provided mount point paths are empty directories which are result of file system mounts.
+	 *
+	 * @param mountPaths List of mount point paths that should exist on file system.
+	 * @param projectPath Project path.
+	 */
+	const expectEmptyMountPoints = (mountPaths, projectPath) => {
+		mountPaths.map((relativePath) => {
+			const fullPath = path.join(projectPath, relativePath);
+
+			expect(fs.existsSync(fullPath)).toBe(true);
+			expect(fs.readdirSync(fullPath)).toEqual([]);
+			expect(fs.lstatSync(fullPath).isDirectory()).toBe(true);
+		});
+	};
+
+	/**
+	 * Expect that all required files exist for PHP.
+	 *
+	 * @param requiredFiles List of files that should be accessible by PHP.
+	 * @param documentRoot Document root of the PHP server.
+	 * @param php NodePHP instance.
+	 */
+	const expectRequiredFiles = (requiredFiles, documentRoot, php) => {
+		requiredFiles.map((relativePath) => {
+			const fullPath = path.join(documentRoot, relativePath);
+			expect(php.fileExists(fullPath)).toBe(true);
+		});
+	};
+
+	/**
+	 * Test that startWPNow in "index", "plugin" and "theme" modes doesn't change anything in the project directory.
+	 */
+	test.each(['index', 'plugin', 'theme'])(
+		'startWPNow starts %s mode',
+		async (mode) => {
+			const exampleProjectPath = path.join(exampleDir, mode);
+			const projectPath = path.join(tmpExampleDirectory, mode);
+
+			const rawOptions: Partial<WPNowOptions> = {
+				projectPath: projectPath,
+			};
+
+			await startWPNow(rawOptions);
+
+			expect(fs.readdirSync(projectPath)).toEqual(
```

Comment:
I think it'd be better to have explicit assertions here, instead of reading data and assigning it to a variable.

Reply:
Sure, but it's still variable data. I think it should be static, otherwise it may break unexpectedly.

Reply:
@danielbachhuber so would you use an array of paths for each `toEqual()` call here, instead of reading example directories?

Reply:
@wojtekn Correct.

Reply:
https://github.com/WordPress/wordpress-playground/pull/359